### ISSUE TITLE
ofDrawBitmapString renders incorrectly using ofGLProgrammableRenderer

### ISSUE THREAD

**Issue by julapy** [openframeworks/openFrameworks#3675]
getting some strange results when using ofDrawBitmapString on ofGLProgrammableRenderer.

`ofDrawBitmapString("testing", 20, 20);` - should just draw a white bitmap string.
instead it draws it on a background and the text is yellow.
![Image](https://dl.dropboxusercontent.com/u/3828359/OF/ofDrawBitmapString_bug0.png)

this is how text labels are rendering using ofxUI.
and i think this might be using a ofTrueTypeFont for the labels.
![Image](https://dl.dropboxusercontent.com/u/3828359/OF/ofDrawBitmapString_bug1.png)

**Comment by tgfrerer** [comment]
Hi Lukasz, I couldn't reproduce this on my system - is there a way to rule out it's something internal to ofxUI?

From the images you posted, could it have something to do with the blue and the alpha channel going missing / not mapping properly?

Also, do you mind posting your configuration? I assume you are using the latest master branch. It would help narrow things a bit down.

**Comment by julapy** [comment]
hey, been doing some more testing and it seems to be related the way im setting up my window.

the below is the old way of creating a window,
which is what is causing these issues.
````
ofGLFWWindowSettings windowSettings;
windowSettings.setGLVersion(4, 1);

ofAppGLFWWindow window;
window.setup(windowSettings);

ofSetupOpenGL(&window, 1024, 768, OF_WINDOW);
ofRunApp( new ofApp());
````

when i create a window the new way,
the issue with the bitmap fonts doesn't happen anymore.
````
ofGLFWWindowSettings windowSettings;
windowSettings.setGLVersion(4, 1);
windowSettings.width = EEPDisplayWidth;
windowSettings.height = EEPDisplayHeight;
windowSettings.windowMode = OF_WINDOW;

ofCreateWindow(windowSettings);
ofRunApp( new ofApp());
````

**Comment by tgfrerer** [comment]
That's great to know! I think we should find a way to enforce the new way of window creation to make things a little more consistent =)

**Comment by arturoc** [comment]
yes, probably just checking that there's not 2 calls to setup the same window, in your example, both `window.setup(settings)` and `ofSetupOpenGL(&window...)` are setting up the window.

**Comment by julapy** [comment]
it would be good if it could still work both ways until the old method is fully deprecated, as i imagine a lot of people will be trying to create windows using the old method and running into this issue.
